### PR TITLE
sigil: update to 1.2.1 and add python variants

### PR DESCRIPTION
**Pull request by i0ntempest**
py-cssutils: add python38 support

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.4 19E266
Xcode 11.4 11E146

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by mf2k** [comment]
The commits should really be one per Portfile.

File: editors/sigil/Portfile [hunk 1]
```diff
@@ -51,9 +38,48 @@ compiler.cxx_standard   2011
...
+                        port:py${python_version}-six
+    configure.env-append PATH=${frameworks_dir}/Python.framework/Versions/${python_branch}/bin:${PATH}
+}
+
+variant python38 conflicts python37 description {Use python38 during build} {
+    set python_branch 3.8
+    python-depends ${python_branch} ${frameworks_dir} $env(PATH)
+}
+
+variant python37 conflicts python38 description {Use python38 during build} {
```
Comment (blazewicz) [inline editors/sigil/Portfile]:
```suggestion
variant python37 conflicts python38 description {Use python37 during build} {
```

File: editors/sigil/Portfile [hunk 2]
```diff
@@ -51,9 +38,48 @@ compiler.cxx_standard   2011
...
+                        port:py${python_version}-cssutils \
+                        port:py${python_version}-html5lib \
+                        port:py${python_version}-lxml \
+                        port:py${python_version}-Pillow \
+                        port:py${python_version}-regex \
+                        port:py${python_version}-six
+    configure.env-append PATH=${frameworks_dir}/Python.framework/Versions/${python_branch}/bin:${PATH}
+}
+
+variant python38 conflicts python37 description {Use python37 during build} {
```
Comment (mf2k) [inline editors/sigil/Portfile]: `python37` here needs to be changed to `python38` to match the variant name.

Comment (i0ntempest) [reply editors/sigil/Portfile]: Done just now

File: editors/sigil/Portfile [hunk 3]
```diff
@@ -51,9 +38,48 @@ compiler.cxx_standard   2011
...
+                        port:py${python_version}-cssutils \
+                        port:py${python_version}-html5lib \
+                        port:py${python_version}-lxml \
+                        port:py${python_version}-Pillow \
+                        port:py${python_version}-regex \
+                        port:py${python_version}-six
+    configure.env-append PATH=${frameworks_dir}/Python.framework/Versions/${python_branch}/bin:${PATH}
+}
+
+variant python38 conflicts python37 description {Use python38 during build} {
```
Comment (mf2k) [inline editors/sigil/Portfile]: Also, I should have mentioned this before. The descriptions should be proper English and not use Macports shortcut names. So Change the description:

`Use python 3.8 during build
`

File: editors/sigil/Portfile [hunk 4]
```diff
@@ -51,9 +38,48 @@ compiler.cxx_standard   2011
...
+    set python_branch 3.8
+    python-depends ${python_branch} ${frameworks_dir} $env(PATH)
+}
+
+variant python37 conflicts python38 description {Use python37 during build} {
```
Comment (mf2k) [inline editors/sigil/Portfile]: Change the description to:

`Use python 3.7 during build`

**Comment by i0ntempest** [comment]
Anyone merge this please?

**Comment by kencu** [comment]
why doesn't Frank merge it? He's looked it over extensively already...

**Comment by i0ntempest** [comment]
I don't know... another pr of mine is also getting no attention.

**Comment by kencu** [comment]
"commit fear". If you don't commit it, you can't be responsible for blowback.

**Comment by i0ntempest** [comment]
Thank you

**Pull request merged**